hub/core/sample.py

```diff
@@ -0,0 +1,117 @@
+from hub.core.compression import compress_array
+from hub.constants import UNCOMPRESSED, USE_UNIFORM_COMPRESSION_PER_SAMPLE
+from hub.core.chunk_engine.flatten import row_wise_to_bytes
+import numpy as np
+import pathlib
+from typing import List, Optional, Tuple
+
+from PIL import Image  # type: ignore
+
+
+class Sample:
+    path: Optional[pathlib.Path]
+
+    def __init__(
+        self,
+        path: str = None,
+        array: np.ndarray = None,
+    ):
+        if (path is None) == (array is None):
+            raise ValueError("Must pass either `path` or `array`.")
+
+        if path is not None:
+            self.path = pathlib.Path(path)
+            self._array = None
+
+        if array is not None:
+            self.path = None
+            self._array = array
+            self._original_compression = UNCOMPRESSED
+
+    @property
+    def is_symbolic(self) -> bool:
```

Comment:
Can we rename this? The name isn't obvious to me. Can it be something like 'lazy' instead?

Reply:
sure